src/lib/redis/cluster.c

```diff
@@ -2539,3 +2543,278 @@ fr_redis_cluster_t *fr_redis_cluster_alloc(TALLOC_CTX *ctx,
 
 	return cluster;
 }
+
+/** Check the requisite number of slaves replicated the lease info
+ *
+ * @param request The current request.
+ * @param wait_num Number of slaves required.
+ * @param reply we got from the server.
+ * @return
+ *	- 0 if enough slaves replicated the data.
+ *	- -1 if too few slaves replicated the data, or another error.
+ */
+static inline int ippool_wait_check(REQUEST *request, uint32_t wait_num, redisReply *reply)
+{
+	if (!wait_num) return 0;
+
+	if (reply->type != REDIS_REPLY_INTEGER) {
+		REDEBUG("WAIT result is wrong type, expected integer got %s",
+			fr_int2str(redis_reply_types, reply->type, "<UNKNOWN>"));
+		return -1;
+	}
+	if (reply->integer < wait_num) {
+		REDEBUG("Too few slaves acknowledged allocation, needed %i, got %lli",
+			wait_num, reply->integer);
+		return -1;
+	}
+	return 0;
+}
+
+int redis_ippool_loadscript_buf(TALLOC_CTX *ctx, redis_ippool_lua_script_t const *preamble, redis_ippool_lua_script_t *script)
+{
+	FILE		*f;
+	struct stat	finfo;
+	int		len;
+	int		preamble_len;
+	char		*pos;
+
+	preamble_len = preamble ? talloc_array_length(preamble->script) : 0;
+
+	f = fopen(script->file, "r");
+	if (!f) {
+		ERROR("Failed opening Redis Lua file \"%s\": %s", script->file,
+		       fr_syserror(errno));
+		goto err;
+	}
+
+	if (fstat(fileno(f), &finfo) < 0) {
+		ERROR("Failed stating Redis Lua file \"%s\": %s", script->file,
+		       fr_syserror(errno));
+		goto err2;
+	}
+
+	if (finfo.st_size > SCRIPT_MAX) {
+		ERROR("Size of Redis Lua (%zu) file (%s) exceeds limit (%uk)",
+		       (size_t) finfo.st_size / 1024, script->file, SCRIPT_MAX / 1024);
+		goto err2;
+	}
+
+	MEM(script->script = talloc_realloc(ctx, script->script, char, preamble_len + finfo.st_size + 1));
+	talloc_set_name_const(script->script, script->file);
+
+	pos = script->script;
+	if (preamble) {
+		memcpy(pos, preamble->script, preamble_len - 1);
+		pos += preamble_len - 1;
+
+		*pos = '\n';
+		pos++;
+	}
+	len = fread(pos, sizeof(char), finfo.st_size, f);
+	if (len != finfo.st_size) {
+		ERROR("Error reading Redis Lua file (%s): %s", script->file, fr_syserror(errno));
```

Comment:
if failed, it should continue or go to some of those `err` labels?

Reply:
Good catch, fixed in https://github.com/jimdigriz/freeradius-server/commit/7a6e9663b1dad9f0c30254763dbbcab7ba5046b2